src/Scene/NodeProcess.js

```diff
@@ -135,6 +197,67 @@ define('Scene/NodeProcess',
         }
     };
 
+    function updateNodeImagery(quadtree, node) {
+        var promises = [];
+
+        for (var i=0; i<quadtree.colorLayers.length; i++) {
+            var layer = quadtree.colorLayers[i];
+
+            if (layer.tileInsideLimit(node, layer)) {
+                var args = {layer: layer, destination: 1 };
+                promises.push(quadtree.interCommand.request(args, node, refinementCommandCancellationFn));
+            }
+        }
+
+        return Promise.all(promises).then(function(colorTextures) {
+            var textures = [];
+            for (var j=0; j<colorTextures.length; j++) {
+                textures = textures.concat(colorTextures[j]);
+            }
+            node.setTexturesLayer(textures, 1);
```

Comment:
`colorTextures` is an array (because of `Promise.all(promises)`) of textures (if the Provider returns a single Promise) or array of textures (is the Provider returns a Promise.all of multiple promises).
Hence the need to concat the various textures in a single array.

Regarding your second suggestion: it could be a good idea. Or at the very least change the name of the function to get rid of 'Layer'